installer/installer.py

```diff
@@ -542,10 +544,14 @@ def setup_cosmovisor(self):
             self.remove_safe("CHANGELOG.md")
             self.remove_safe("README.md")
             self.remove_safe("LICENSE")
-            self.mkdir_p(self.cosmovisor_root_dir)
-            self.mkdir_p(os.path.join(self.cosmovisor_root_dir, "genesis"))
-            self.mkdir_p(os.path.join(self.cosmovisor_root_dir, "genesis/bin"))
-            self.mkdir_p(os.path.join(self.cosmovisor_root_dir, "upgrades"))
+            
+            if not is_cosmovisor_bump_needed:
```

Comment:
Remember that `setup_cosmovisor` is **also** used for new/setup from scratch. Don't assume that this logic needs to execute only if a bump is needed.

I'd keep the logic of `setup_cosmovisor` exactly as it is now since it covers the install from scratch, or in upgrade scenario where the user chooses yes to install from scratch use case. Split out the function for whether a Cosmovisor bump is needed to its own function, that way you don't overload this function.